App/Redux/TextileRedux.js

```diff
@@ -43,7 +46,7 @@ export const INITIAL_STATE = Immutable({
 export const TextileSelectors = {
   // TODO: Add more selectors here as we learn how they are used
   itemsById: (state, id) => {
-    return state.textile.images.items.filter(item => item.id === id)
+    return state.textile.images.items.filter(item => item.hash === id)
```

Comment:
Set the custom upload id to the hash, so we no longer need the reducer that persists the id.